Assets/Scripts/ChessPiece.cs

```diff
@@ -192,10 +197,12 @@ private void OnSpeedUpgradePressed(PointerEventData _)
         if (_currencyManager.Value.TryRemoveCurrency(GlobalGameAssets.Instance.CurrencyBalanceData.UpgradeSpeedCost) )
         {
             Speed++;
+            OnSuccessfulUpgrade.Invoke();
```

Comment:
When invoking actions, its good practice to always do a null check so it does not invoke if there are no subscribers.
We do this by: 

`OnSuccessfulUpgrade?.Invoke();`

just add the '?' before the .